# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/Element.cpp
# Commit: 3508e2bd4e3c
# Full Hash: 3508e2bd4e3cbc756cff7da7aee3c60c29c15bd1
# Author: Frédéric Wang <fwang@igalia.com>
# Date: 2024-02-29 21:06:42
# Regressor Bug: 1807253
# File Overlap Count: 1
# Description:
#   Bug 1880928 - Remove last remembered size observer, handle it in Document::DetermineProximityToViewportAndNotifyResizeObservers. r=emilio,layout-reviewers
#   
#   The CSS Box Sizing specification indicates that last remembered sizes
#   are recorded "at the time that ResizeObserver events are determined and
#   delivered" [1].
# ==============================================================================

diff -r 467f563c9ea5 -r 3508e2bd4e3c dom/base/Element.cpp
--- a/dom/base/Element.cpp	Thu Feb 29 08:50:55 2024 +0000
+++ b/dom/base/Element.cpp	Thu Feb 29 08:50:55 2024 +0000
@@ -2144,11 +2144,10 @@
     }
 
     if (HasLastRememberedBSize() || HasLastRememberedISize()) {
-      // Need to remove the last remembered size at the next ResizeObserver
-      // opportunity, so observe the element. But if already observed, we still
-      // want the callback to be invoked even if the size was already 0x0, so
-      // unobserve it first.
-      document->UnobserveForLastRememberedSize(*this);
+      // Make sure the element is observed so that remembered sizes are kept
+      // until the next time "ResizeObserver events are determined and
+      // delivered". See "Disconnected element" tests from
+      // css/css-sizing/contain-intrinsic-size/auto-006.html
       document->ObserveForLastRememberedSize(*this);
     }
   }